### PR TITLE
feat: add REST API endpoints for manipulation of Access Form elements and sections

### PR DESCRIPTION
**Pull request by RadovanTomik**
### Description:

Aadd REST API endpoints for manipulation of Access Form elements and sections. To test them please use swagger.

### Checklist:

_Make sure you tick all the boxes bellow if they are true or do not apply:_

- [x] I have performed a self review of my code
- [x] My code follows [Google Java Code style](https://google.github.io/styleguide/javaguide.html)
- [x] I have made my code as simple as possible
- [x] I have added unit tests and the code coverage has not decreased
- [x] I have updated the documentation in all relevant places

### CODE REVIEW

File: src/main/java/eu/bbmri_eric/negotiator/api/controller/v3/AccessFormController.java [hunk 1]
```diff
@@ -1,37 +1,76 @@
...
 
-  @Autowired private AccessCriteriaSetService accessCriteriaSetService;
-  @Autowired private AccessFormService accessFormService;
-  @Autowired private AccessFormModelAssembler accessFormModelAssembler;
+  private final AccessCriteriaSetService accessCriteriaSetService;
+  private final AccessFormElementService elementService;
+  private final AccessFormsSectionService sectionService;
+  private final AccessFormService accessFormService;
+  private final AccessFormModelAssembler accessFormModelAssembler;
+  private final ModelMapper modelMapper;
```
Comment (svituz) [inline src/main/java/eu/bbmri_eric/negotiator/api/controller/v3/AccessFormController.java:50]: modelMapper is not used in the class. Only assigned in the constructor

File: src/main/java/eu/bbmri_eric/negotiator/api/controller/v3/AccessFormController.java [hunk 2]
```diff
@@ -1,37 +1,76 @@
...
 import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestController;
 
 @RestController
-@RequestMapping("/v3")
-@CrossOrigin
+@RequestMapping(
+    value = "/v3",
+    produces = MediaTypes.HAL_JSON_VALUE,
+    consumes = MediaType.APPLICATION_JSON_VALUE)
```
Comment (svituz) [inline src/main/java/eu/bbmri_eric/negotiator/api/controller/v3/AccessFormController.java:41]: This is causing an exception when getting the negotiations. 
Using postman to GET /api/v3/access-forms it results in 
```
{
    timestamp": "2024-04-23",
    "status": 415,
    "error": "Unsupported Media Type",
    "trace": "org.springframework.web.HttpMediaTypeNotSupportedException: Content-Type ....
}
```
I guess it expects json body that the GET call doesn't need to send

Comment (RadovanTomik) [reply src/main/java/eu/bbmri_eric/negotiator/api/controller/v3/AccessFormController.java:41]: yes, fixed in the next PR but I can remove it here as well

File: src/main/java/eu/bbmri_eric/negotiator/dto/access_form/ElementCreateDTO.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+@Builder
+@JsonInclude(value = JsonInclude.Include.NON_NULL)
+public class ElementCreateDTO {
+  @NotNull private String name;
+
+  @NotNull private String label;
+
+  @NotNull private String description;
+
+  @NotNull private String type;
```
[the remark this answers is not in the document]
Comment (RadovanTomik) [reply src/main/java/eu/bbmri_eric/negotiator/dto/access_form/ElementCreateDTO.java:24]: yes, I plan to add it in the next PR that will deal with new element types

File: src/main/java/eu/bbmri_eric/negotiator/dto/access_form/SectionCreateDTO.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import lombok.Setter;
+
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+@JsonInclude(value = JsonInclude.Include.NON_NULL)
+public class SectionCreateDTO {
+  @NotNull private String name;
```
[the remark this answers is not in the document]
Comment (RadovanTomik) [reply src/main/java/eu/bbmri_eric/negotiator/dto/access_form/SectionCreateDTO.java:18]: hmm... does it matter if it has spaces or not?

File: src/main/java/eu/bbmri_eric/negotiator/dto/access_form/ElementMetaDTO.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+@JsonInclude(value = JsonInclude.Include.NON_NULL)
+@Relation(collectionRelation = "elements", itemRelation = "element")
+public class ElementMetaDTO {
+  @NotNull private Long id;
+  @NotNull private String name;
```
[the remark this answers is not in the document]
Comment (RadovanTomik) [reply src/main/java/eu/bbmri_eric/negotiator/dto/access_form/ElementMetaDTO.java:21]: same as above

**Review by svituz — APPROVED** [review]

**Pull request merged**